client/rpc/key.go

```diff
@@ -141,3 +143,53 @@ func (api *KeyAPI) Remove(ctx context.Context, name string) (iface.Key, error) {
 func (api *KeyAPI) core() *HttpApi {
 	return (*HttpApi)(api)
 }
+
+func (api *KeyAPI) Sign(ctx context.Context, name string, data []byte) (iface.Key, []byte, error) {
+	var out struct {
+		Key       keyOutput
+		Signature string
+	}
+
+	err := api.core().Request("key/sign").
+		Option("key", name).
+		FileBody(bytes.NewReader(data)).
+		Exec(ctx, &out)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	key, err := newKey(out.Key.Name, out.Key.Id)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	_, signature, err := multibase.Decode(out.Signature)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return key, signature, nil
+}
+
+func (api *KeyAPI) Verify(ctx context.Context, keyOrName string, signature, data []byte) (iface.Key, bool, error) {
+	var out struct {
+		Key            keyOutput
+		SignatureValid bool
+	}
+
+	err := api.core().Request("key/verify").
+		Option("key", keyOrName).
+		Option("signature", toMultibase(signature)).
```

Comment:
Could we devise a human readable combined scheme ?
`$PEERID-$SIG` ?
Can't use PEERID for RSA but else should be fine.

Reply:
Do you mean the signature to include the peer ID such that we do not need to provide the key?

Reply:
Yes, forget the ECC thing it's not what we are trying to do right now.

Instead of `-key Qmfoo -sig zMM...` `Qmfoo-zMM` ?

Reply:
I don't understand what you mean, if it already contains the key, why do I need to specify the peer id ?

Reply:
We want this API to accept `key` as a pet name (`ipfs key list`) or the actual libp2p-key (peerid).

Combining key and sig into a single arg makes the meaning of `mfoo-mbar-mbuz` ambiguous because we allow `-` in pet names AND mbase64url:
- is the key name `mfoo` or `mfoo-mbar`?
- is the signature `mbar-mbuz` or `mbuz`? (mind that `-` is part of base64url alphabet)